python/pyspark/sql/dataframe.py

```diff
@@ -1873,11 +1873,11 @@ def sample(  # type: ignore[misc]
         Examples
         --------
         >>> df = spark.range(10)
-        >>> df.sample(0.5, 3).count()
+        >>> df.sample(0.5, 3).count() # doctest: +SKIP
         7
-        >>> df.sample(fraction=0.5, seed=3).count()
+        >>> df.sample(fraction=0.5, seed=3).count() # doctest: +SKIP
         7
-        >>> df.sample(withReplacement=True, fraction=0.5, seed=3).count()
+        >>> df.sample(withReplacement=True, fraction=0.5, seed=3).count() # doctest: +SKIP
```

Comment:
The result is deterministic as long as 1. seed is same, and 2. the number of partitions in the originated DataFrame. I agree with the change in a way that you might want to run this doctest with a different env but Python version shouldn't be a valid reason for it. Please either elaborate it, or fix the PR description. LGTM otherwise.